runbotics/runbotics-orchestrator-ui/src/src-landing/components/LanguageSwitcher/LanguageSwitcher.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { VFC, useEffect } from 'react';
+
+import { useRouter } from 'next/router';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+import { languages, Language } from '#src-app/translations/translations';
+
+import { capitalizeFirstLetter } from '../../utils/utils';
+
+import styles from './LanguageSwitcher.module.scss';
+
+const LanguageSwitcher: VFC = () => {
+    const { switchLanguage, translate } = useTranslations();
+
+    const { push, locale: activeLocale, asPath } = useRouter();
+
+    const handleLanguageSwitch = (language: Language) => {
+        push(asPath, null, {
+            locale: language,
+        });
+    };
+
+    useEffect(() => {
+        switchLanguage(activeLocale as Language);
+        push(asPath, undefined, {
+            locale: activeLocale,
+        });
```

Comment:
not sure if we need those 2. Imo push should adjust locale in i18n. Check it plx